### PR TITLE
features/Tutorial

### PR DESCRIPTION
**Pull request by zhaoan12**
Updated javadoc on GiveTutorial entity. Create a new use cases class that tracks the current "phase" of the tutorial, whether the tutorial is completed, and is able to change the status of different tutorial actions. Created test cases for multiple methods in the class. Added a Utils class that contains Strings for the tutorial.

### CODE REVIEW

File: src/main/java/com/mg105/use_cases/PlayerGetsTutorial.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Check if all required actions performed by player
+     */
+    public boolean isComplete(){
+        return tutorial.ActionPerformedGetter("moved") & tutorial.ActionPerformedGetter("attacked")
+            & tutorial.ActionPerformedGetter("usedItem");
+    }
+
+    /**
+     * Get current phase of tutorial, which is first index of phase list
```
Comment (sbrad044) [inline src/main/java/com/mg105/use_cases/PlayerGetsTutorial.java:29]: why/how is the current phase always the first index?

File: src/main/java/com/mg105/entities/GiveTutorial.java [hunk 2]
```diff
@@ -7,6 +7,12 @@ public class GiveTutorial {
     private boolean attacked = false;
     private boolean usedItem = false;
 
+    public GiveTutorial(boolean moved, boolean attacked, boolean usedItem){
```
Comment (sbrad044) [inline src/main/java/com/mg105/entities/GiveTutorial.java:10]: why not just have these variables set to false by default? unless there's a scenario in which the user can redo certain parts of the tutorial I guess. I assumed the tutorial would be a one time thing

**Review by sbrad044 — APPROVED** [review]

File: src/main/java/com/mg105/use_cases/PlayerGetsTutorial.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Action has been performed, so set it to true
+     */
+    public void setActionPerformed(String action){
+        this.tutorial.ActionPerformedSetter(action);
+    }
+
+    /**
+     * Check if specific action has been performed
+     */
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/use_cases/PlayerGetsTutorial.java:44]: Javadoc is incorrect. There should be at least one `@param` for the missing argument.

**Review by 6167656e74323431 — CHANGES_REQUESTED** [review]
In addition to the comments the autoformatter clearly hasn't been run. Use it please.

Content-wise it seems fine.

File: src/main/java/com/mg105/use_cases/PlayerGetsTutorial.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        this.tutorial = tutorial;
+    }
+
+    /**
+     * Check if all required actions performed by player
+     */
+    public boolean isComplete(){
+        return tutorial.ActionPerformedGetter("moved") & tutorial.ActionPerformedGetter("attacked")
+            & tutorial.ActionPerformedGetter("usedItem");
+    }
```
Comment (siddharthgowda) [inline src/main/java/com/mg105/use_cases/PlayerGetsTutorial.java:26]: It seems redundant to have the completed attribute and having this method as well. Also these values are probably better off put as constants although that really isn't a big deal.

**Review by siddharthgowda — CHANGES_REQUESTED** [review]
Overall things seem pretty good. Some of the things in the usecase could also have been placed in the entity layer (not saying that putting it in the usecase layer was wrong, it's just another option). Also look at my comment.

**Pull request closed without merging**